pnc-rest/src/main/java/org/jboss/pnc/rest/endpoint/ProjectEndpoint.java

```diff
@@ -59,8 +62,10 @@ public Response createNew(@NotNull @Valid ProjectRest projectRest, @Context UriI
     @ApiOperation(value = "Updates an existing Project")
     @PUT
     @Path("/{id}")
-    public Response update(@ApiParam(value = "Project id", required = true) @PathParam("id") Integer productId,
+    public Response update(@ApiParam(value = "Project id", required = true) @PathParam("id") Integer id,
             @NotNull @Valid ProjectRest projectRest, @Context UriInfo uriInfo) {
+        validateIdIsNull(projectRest);
```

Comment:
Actually I think in that case we would need 2 validation annotations. One for validating if `id` is present and the second one for validating if it's missing.